example/src/main/java/io/netty/example/ocsp/OcspClientExample.java

```diff
@@ -0,0 +1,224 @@
+import io.netty.handler.codec.http.FullHttpRequest;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpClientCodec;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpMethod;
+import io.netty.handler.codec.http.HttpObjectAggregator;
+import io.netty.handler.codec.http.HttpVersion;
+import io.netty.handler.ssl.OpenSsl;
+import io.netty.handler.ssl.ReferenceCountedOpenSslContext;
+import io.netty.handler.ssl.ReferenceCountedOpenSslEngine;
+import io.netty.handler.ssl.SslContextBuilder;
+import io.netty.handler.ssl.SslHandler;
+import io.netty.handler.ssl.SslProvider;
+import io.netty.handler.ssl.ocsp.ClientOcspContext;
+import io.netty.util.ReferenceCountUtil;
+import io.netty.util.concurrent.Promise;
+
+/**
+ * This is a very simple example for a HTTPS client that uses OCSP stapling.
+ * The client connects to a HTTPS server that has OCSP stapling enabled and
+ * then uses BC to parse and validate it.
+ */
+public class OcspClientExample {
+    public static void main(String[] args) throws Exception {
+        if (!OpenSsl.isAvailable()) {
+            throw new IllegalStateException("OpenSSL is not available!");
+        }
+
+        if (!OpenSsl.isOcspSupported()) {
+            throw new IllegalStateException("OCSP is not supported!");
+        }
+
+        // Using Wikipedia as an example. I'd rather use Netty's own website
+        // but the server doesn't support OCSP. A few other examples could
+        // be Microsoft or Squarespace. Use OpenSSL's CLI client to assess
+        // if a server supports OCSP stapling. E.g.:
+        //
+        // openssl s_client -tlsextdebug -status -connect www.squarespace.com:443
+        //
+        final String host = "www.wikipedia.org";
+
+        ReferenceCountedOpenSslContext context
+            = (ReferenceCountedOpenSslContext) SslContextBuilder.forClient()
+                .sslProvider(SslProvider.OPENSSL)
+                .build();
+        try {
+            context.ocspContext(new ClientOcspContext() {
+                @Override
+                public boolean verify(byte[] staple) throws Exception {
+                    if (staple == null) {
+                        throw new IllegalStateException("Server didn't provide an OCSP staple!");
+                    }
+
+                    OCSPResp response = new OCSPResp(staple);
+                    if (response.getStatus() != OCSPResponseStatus.SUCCESSFUL) {
+                        return false;
+                    }
+
+                    BasicOCSPResp basicResponse = (BasicOCSPResp) response.getResponseObject();
+                    SingleResp first = basicResponse.getResponses()[0];
+
+                    CertificateStatus status = first.getCertStatus();
+                    System.out.println(host);
+                    System.out.println("  Status: " + (status == CertificateStatus.GOOD ? "Good" : status));
```

Comment:
consider putting this in a string building to increase the likelihood this string prints out in a contiguous manner. This would be more consistent with other examples and self contained.